R/utils.R

```diff
@@ -101,6 +124,10 @@ clear_py_env <- function() {
   return(invisible())
 }
 
+local_py_env <- function(envir = parent.frame()) {
+  withr::defer(py_clear_env())
```

Comment:
This one is important for ensuring that `local_py_env()` clears when exiting the parent scope rather than immediately.

```suggestion
  withr::defer(py_clear_env(), envir = envir)
```

Reply:
Ah missed that one, thanks!